src/main/java/com/gradle/GitMetadata.java

```diff
@@ -0,0 +1,42 @@
+package com.gradle;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.gradle.Utils.execAndGetStdOut;
+
+class GitMetadata {
+
+    private String[] args;
+    boolean isGitInstalled;
+    private Supplier<Optional<String>> fromEnv;
+
+    private GitMetadata(boolean isGitInstalled) {
+        this.isGitInstalled = isGitInstalled;
+    }
+
+    static GitMetadata given(boolean isGitInstalled) {
+        return new GitMetadata(isGitInstalled);
+    }
+
+    GitMetadata fromCmd(String... args) {
```

Comment:
it seems that this class is very close to the standard Java Builder pattern. the `fromCmd` is also a required parameter. can we adjust this class to use the standard Java Builder pattern where this is a requirement?

Reply:
good point, see latest changes, I think it looks better. There's a hidden trap there: since I keep the builder around for simplicity sake, you have to be careful not to re-use the previous state. On the other hand re-creating the builder every time seems overkill...